sdks/go/pkg/beam/runners/prism/internal/engine/elementmanager.go

```diff
@@ -532,12 +533,26 @@ func (em *ElementManager) StateForBundle(rb RunBundle) TentativeData {
 	return ret
 }
 
+// Residual represents the unprocessed portion of a single element to be rescheduled for processing later.
+type Residual struct {
+	Element []byte
+	Delay   time.Duration // The relative time delay.
+	Bounded bool          // Whether this element is finite or not.
+}
+
+// Residuals is used to specify process continuations within a bundle.
+type Residuals struct {
```

Comment:
Could we say instead, including the use of a symbol link,: "Residuals stores remaining [Residual]s, their aggregated Watermarks, and references needed to query their associated transforms"

Question triggered for me by "process continuations" that relates to me in Java where the ProcessContinuation is the return type for all ProcessElement lifecycle methods, whether or not the developer uses the void return type:
Is `Residuals` instantiated even when a DoFn does not explicitly set a process timer? In other words, do the code paths converge to using Residuals, both when remaining elements exist in a bundle and when process timers are explicitly configured for a DoFn?

Reply:
Sure. I haven't tried using the go 1.19 godoc improvements yet. But be aware these are internal packages and it's very unlikely that someone will ever look at the godoc for them without simply looking at the code directly.

I'll do this in a subsequent PR since this change is pre-work, and far from its final shape.

-----

Short answer: No. It's not instantiated in any meaningful way if the user code doesn't return a ProcessContinuation. Delay or similar, since SDK doesn't return residuals if the user doesn't explicitly use a ProcessContinuation.

Timers are an orthogonal more primitive concept. One could implement processing continuations in terms of ProcessingTime timers and State. That might be how this ultimately ends up working. I don't know yet.

Residuals is a type, so to talk about instantiation you need to point to a specific instance or not. It's also a small type with 4 fields. If the map and slices remain instantiated themselves, then they are nil. Go struct instantiations are very very cheap.